onnxmltools/convert/common/_topology.py

```diff
@@ -233,7 +232,8 @@ def delete_local_variable(self, onnx_name):
 class Topology:
 
     def __init__(self, model, default_batch_size=1, initial_types=None,
-                 reserved_variable_names=None, reserved_operator_names=None, targeted_onnx=None):
+                 reserved_variable_names=None, reserved_operator_names=None, targeted_onnx=None,
+                 custom_conversion_functions={}, custom_shape_calculators={}):
```

Comment:
Please set default to `None` and change it to `{}` inside this constructor's body. Note that the initial value are shared by ALL calls to this function, so we cannot use classes here.

Reply:
I didn't get your point, it should be ok for sharing among its object since it is read-only in the object of Topology.

Reply:
Thanks for the reminder, very tricky thing that I firstly heard of.